<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Database of anor â€¢ anor</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="../life.css" rel="stylesheet">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- life --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">anor</a>
        <small class="life">developed by <a href="http://life2cloud.com">Jianfeng</a></small>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li>
  <a href="../articles/introduction_to_anor.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/database_in_anor.html">Database in anor</a>
    </li>
    <li>
      <a href="../articles/shiny_of_anor.html">Shiny app of anor</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
        <li>
  <a href="https://github.com/JhuangLab/anor">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="database_of_anor_files/htmlwidgets-1.5.1/htmlwidgets.js"></script><script src="database_of_anor_files/jquery-1.12.4/jquery.min.js"></script><link href="database_of_anor_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="database_of_anor_files/datatables-binding-0.13/datatables.js"></script><link href="database_of_anor_files/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="database_of_anor_files/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="database_of_anor_files/dt-core-1.10.20/js/jquery.dataTables.min.js"></script><link href="database_of_anor_files/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet">
<script src="database_of_anor_files/crosstalk-1.1.0.1/js/crosstalk.min.js"></script><link href="database_of_anor_files/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet">
<script src="database_of_anor_files/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script><script src="database_of_anor_files/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script><script src="database_of_anor_files/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script><script src="database_of_anor_files/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script><script src="database_of_anor_files/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script><link href="database_of_anor_files/dt-ext-fixedcolumns-1.10.20/css/fixedColumns.dataTables.min.css" rel="stylesheet">
<script src="database_of_anor_files/dt-ext-fixedcolumns-1.10.20/js/dataTables.fixedColumns.min.js"></script><link href="database_of_anor_files/dt-ext-responsive-1.10.20/css/responsive.dataTables.min.css" rel="stylesheet">
<script src="database_of_anor_files/dt-ext-responsive-1.10.20/js/dataTables.responsive.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Database of anor</h1>
                        <h4 class="author">Jianfeng Li</h4>
            
            <h4 class="date">2020-04-20</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/JhuangLab/anor/blob/master/vignettes/database_of_anor.Rmd"><code>vignettes/database_of_anor.Rmd</code></a></small>
      <div class="hidden name"><code>database_of_anor.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>In this reference manual, we arranged the databases in anor and provide several meta information about the anor supported databases and other BioInstaller supported (download only) databases. Some of description or comments have been described in <a href="https://github.com/JhuangLab/BioInstaller/blob/master/inst/extdata/config/db/db_annovar.toml">the download configuration file</a> (BioInstaller package) and <a href="https://github.com/JhuangLab/anor/blob/master/inst/extdata/config/databases.toml">the annotation configuration file</a> (anor).</p>
<p>anor supported databases will firstly be supported to download from original sites (exclude the authentication part) in BioInstaller. A portion of BioInstaller supported databases will be introduced in anor to as the candidate annotation databases (process method: remain unchanged, re-formate, re-analysis).</p>
</div>
<div id="overview-of-supported-database" class="section level2">
<h2 class="hasAnchor">
<a href="#overview-of-supported-database" class="anchor"></a>Overview of supported database</h2>
<p>The followed table shows all annotation names with its required download name.</p>
<div id="htmlwidget-c9157eef89c557fe9b08" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c9157eef89c557fe9b08">{"x":{"filter":"none","caption":"<caption>Download name of anor database.<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205"],["1000g2015aug_all","1000g2015aug_afr","1000g2015aug_amr","1000g2015aug_eas","1000g2015aug_eur","1000g2015aug_sas","perl_annovar_refGene","perl_annovar_ensGene","perl_annovar_knownGene","perl_annovar_cytoBand","perl_annovar_genomicSuperDups","perl_annovar_clinvar_20170905","perl_annovar_esp6500siv2_all","perl_annovar_1000g2015aug_all","perl_annovar_1000g2015aug_afr","perl_annovar_1000g2015aug_eas","perl_annovar_1000g2015aug_eur","perl_annovar_avsnp142","perl_annovar_avsnp144","perl_annovar_avsnp147","perl_annovar_avsnp150","perl_annovar_dbnsfp33a","perl_annovar_dbnsfp35a","perl_annovar_dbnsfp35c","perl_annovar_cosmic70","perl_annovar_cosmic81","perl_annovar_cosmic82","!!glue {paste0('perl_annovar_cosmic', 83:88, '_coding')}","perl_annovar_eigen","perl_annovar_gwava","perl_annovar_cadd13","perl_annovar_cadd13gt10","perl_annovar_cadd13gt20","perl_annovar_regsnpintron","perl_annovar_mitimpact24","perl_annovar_popfreq_all_20150413","perl_annovar_popfreq_max_20150413","perl_annovar_icgc21","perl_annovar_nci60","perl_annovar_gme","perl_annovar_hrcr1","perl_annovar_kaviar_20150923","perl_annovar_gnomad_genome","perl_annovar_gnomad_exome","perl_annovar_exac03nonpsych","perl_annovar_exac03nontcga","perl_annovar_exac03","perl_annovar_cg69","perl_annovar_cg46","perl_annovar_intervar_20170202","perl_annovar_intervar_20180118","perl_annovar_dbscsnv11","perl_annovar_merge","avsnp138","avsnp142","avsnp144","avsnp147","avsnp150","bioc_org_hs_eg","bioc_gene2","bioc_gene2alias","bioc_gene2ensembl_gene_id","bioc_gene2entrez_gene_id","bioc_gene2gene_full_name","brvar_v1_core","brvar_v1_extra","cadd","caddgt10","caddgt20","cadd13","cadd13gt10","cadd13gt20","cancer_hotspots_v2","nightly_civic_gene_summaries","clinvar_20170130","clinvar_20170501","clinvar_20170905","clinvar_20180603","clinvar_20190305","cosmic70","cosmic81","cosmic82","!!glue {paste0('cosmic', 83:88, '_coding')}","cscd_cancer_circrna","cscd_common_circrna","cscd_normal_circrna","darned","dbnsfp30a","dbnsfp31a_interpro","dbnsfp33a","dbscsnv11","dhs_gene_connectivity_2012","disgenet_gene_disease","disgenet_befree_gene_disease","disgenet_curated_gene_disease","disgenet_rs_disease","disgenet_befree_rs_disease","disgenet_curated_rs_disease","disgenet_pubannotator_rs_disease","docm_3.2","eigen","epi_genes_v1","exac03","exac03nontcga","exac03nonpsych","gdi_score_full_10282015","gme","gnomad_exome","gnomad_genome","gnomad211_exome","gnomad211_genome","gtex_adipose_subcutaneous.v7.egenes","gtex_adipose_visceral_omentum.v7.egenes","gtex_adrenal_gland.v7.egenes","gtex_artery_aorta.v7.egenes","gtex_artery_coronary.v7.egenes","gtex_artery_tibial.v7.egenes","gtex_brain_amygdala.v7.egenes","gtex_brain_anterior_cingulate_cortex_ba24.v7.egenes","gtex_brain_caudate_basal_ganglia.v7.egenes","gtex_brain_cerebellar_hemisphere.v7.egenes","gtex_brain_cerebellum.v7.egenes","gtex_brain_cortex.v7.egenes","gtex_brain_frontal_cortex_ba9.v7.egenes","gtex_brain_hippocampus.v7.egenes","gtex_brain_hypothalamus.v7.egenes","gtex_brain_nucleus_accumbens_basal_ganglia.v7.egenes","gtex_brain_putamen_basal_ganglia.v7.egenes","gtex_brain_spinal_cord_cervical_c_1.v7.egenes","gtex_brain_substantia_nigra.v7.egenes","gtex_breast_mammary_tissue.v7.egenes","gtex_cells_ebv_transformed_lymphocytes.v7.egenes","gtex_cells_transformed_fibroblasts.v7.egenes","gtex_colon_sigmoid.v7.egenes","gtex_colon_transverse.v7.egenes","gtex_esophagus_gastroesophageal_junction.v7.egenes","gtex_esophagus_mucosa.v7.egenes","gtex_esophagus_muscularis.v7.egenes","gtex_heart_atrial_appendage.v7.egenes","gtex_heart_left_ventricle.v7.egenes","gtex_liver.v7.egenes","gtex_lung.v7.egenes","gtex_minor_salivary_gland.v7.egenes","gtex_muscle_skeletal.v7.egenes","gtex_nerve_tibial.v7.egenes","gtex_ovary.v7.egenes","gtex_pancreas.v7.egenes","gtex_pituitary.v7.egenes","gtex_prostate.v7.egenes","gtex_skin_not_sun_exposed_suprapubic.v7.egenes","gtex_skin_sun_exposed_lower_leg.v7.egenes","gtex_small_intestine_terminal_ileum.v7.egenes","gtex_spleen.v7.egenes","gtex_stomach.v7.egenes","gtex_testis.v7.egenes","gtex_thyroid.v7.egenes","gtex_uterus.v7.egenes","gtex_vagina.v7.egenes","gtex_whole_blood.v7.egenes","gwava","hgnc_coding_gene2alias","hgnc_coding_pre2gene","hrcr1","icgc21","intervar_20170202","intervar_20180118","intogen_20180119","kaviar_20150923","lncediting","loftool_scores","mcap","nci60","2016sih_wes_ball","2016sih_wes_tall","2016sih_wes_nkt","2016sih_wgs_nkt","2016sih_wgs_dlbcl","2018sih_wgs_nkt_full","2018sih_wgs_nkt_full_sqlite","omim_genemap2_ensembl_gene_id2phenotype","omim_genemap2_entrez_gene_id2phenotype","omim_genemap2_symbol2phenotype","popfreq_max_20150413","popfreq_all_20150413","radar2","rddpred_mes","REDIportal","ucsc_refgene","ens_refgene","revel","rs2pos138","rs2pos142","rs2pos144","rs2pos147","rs2pos150","rvis_exac_4kw","seeqtl_qvalue_hapmap3_cis","seeqtl_qvalue_hapmap3_trans","seeqtl_qvalue_myers_cis","seeqtl_qvalue_myers_trans","tall_somatic_genes_20171206","tmcsnpdb","varcards","vcfanno_demo","vep_all"],["db_annovar_1000g_sqlite","db_annovar_1000g_sqlite","db_annovar_1000g_sqlite","db_annovar_1000g_sqlite","db_annovar_1000g_sqlite","db_annovar_1000g_sqlite","db_annovar_refGene","db_annovar_ensGene","db_annovar_knownGene","db_annovar_cytoBand","db_annovar_genomicSuperDups","db_annovar_clinvar_20170905","db_annovar_esp6500siv2_all","db_annovar_1000g2015aug_all","db_annovar_1000g2015aug_afr","db_annovar_1000g2015aug_eas","db_annovar_1000g2015aug_eur","db_annovar_avsnp142","db_annovar_avsnp144","db_annovar_avsnp147","db_annovar_avsnp150","db_annovar_dbnsfp33a","db_annovar_dbnsfp35a","db_annovar_dbnsfp35c","db_annovar_cosmic70","db_annovar_cosmic81","db_annovar_cosmic82","!!glue {paste0('db_annovar_cosmic', 83:88, '_coding')}","db_annovar_eigen","db_annovar_gwava","db_annovar_cadd13","db_annovar_cadd13gt10","db_annovar_cadd13gt20","db_annovar_regsnpintron","db_annovar_mitimpact24","db_annovar_popfreq_all_20150413","db_annovar_popfreq_max_20150413","db_annovar_icgc21","db_annovar_nci60","db_annovar_gme","db_annovar_hrcr1","db_annovar_kaviar_20150923","db_annovar_gnomad_genome","db_annovar_gnomad_exome","db_annovar_exac03nonpsych","db_annovar_exac03nontcga","db_annovar_exac03","db_annovar_cg69","db_annovar_cg46","db_annovar_intervar_20170202","db_annovar_intervar_20180118","db_annovar_dbscsnv11","*","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_bioc_org.Hs.eg.db","db_bioc_org.Hs.eg.db","db_bioc_org.Hs.eg.db","db_bioc_org.Hs.eg.db","db_bioc_org.Hs.eg.db","db_bioc_org.Hs.eg.db","db_annovar_brvar","db_annovar_brvar","db_annovar_cadd_sqlite","db_annovar_cadd_sqlite","db_annovar_cadd_sqlite","db_annovar_cadd_sqlite","db_annovar_cadd_sqlite","db_annovar_cadd_sqlite","db_annovar_cancer_hotspots","db_annovar_civic_gene_summaries","db_annovar_clinvar_sqlite","db_annovar_clinvar_sqlite","db_annovar_clinvar_sqlite","db_annovar_clinvar_sqlite","db_annovar_clinvar_sqlite","db_annovar_cosmic_sqlite","db_annovar_cosmic_sqlite","db_annovar_cosmic_sqlite","db_annovar_cosmic_sqlite","db_annovar_cscd_sqlite","db_annovar_cscd_sqlite","db_annovar_cscd_sqlite","db_annovar_darned_sqlite","db_annovar_dbnsfp_sqlite","db_annovar_dbnsfp_sqlite","db_annovar_dbnsfp_sqlite","db_annovar_dbscsnv_sqlite","db_annovar_dhs_gene_connectivity","db_annovar_disgenet","db_annovar_disgenet","db_annovar_disgenet","db_annovar_disgenet","db_annovar_disgenet","db_annovar_disgenet","db_annovar_disgenet","db_annovar_docm","db_annovar_eigen_sqlite","db_annovar_epi_genes","db_annovar_exac03_sqlite","db_annovar_exac03_sqlite","db_annovar_exac03_sqlite","db_annovar_gdi_score","db_annovar_gme_sqlite","db_annovar_gnomad_sqlite","db_annovar_gnomad_sqlite","db_annovar_gnomad_sqlite","db_annovar_gnomad_sqlite","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gwava_sqlite","db_annovar_hgnc","db_annovar_hgnc","db_annovar_hrcr1_sqlite","db_annovar_icgc_sqlite","db_annovar_intervar_sqlite","db_annovar_intervar_sqlite","db_annovar_intogen_sqlite","db_annovar_kaviar_sqlite","db_annovar_lncediting_sqlite","db_annovar_loftool_scores","db_annovar_mcap_sqlite","db_annovar_nci60","db_annovar_normal_pool","db_annovar_normal_pool","db_annovar_normal_pool","db_annovar_normal_pool","db_annovar_normal_pool","db_annovar_normal_pool","db_annovar_normal_pool_sqlite","db_annovar_omim_genemap2","db_annovar_omim_genemap2","db_annovar_omim_genemap2","db_annovar_popfreq_sqlite","db_annovar_popfreq_sqlite","db_annovar_radar_sqlite","db_annovar_rddpred_sqlite","db_annovar_rediportal_sqlite","db_annovar_refgene","db_annovar_refgene","db_annovar_revel_sqlite","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_annovar_rvis_esv_score","db_annovar_seeqtl","db_annovar_seeqtl","db_annovar_seeqtl","db_annovar_seeqtl","db_annovar_tall_somatic_genes","db_annovar_tmcsnpdb","db_annovar_varcards_sqlite","db_vcfanno_demo","db_vep"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>anno.name<\/th>\n      <th>download.name<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script><p>The followed table shows all database names with its versions and description.</p>
<div id="htmlwidget-5b159aceb12908a46702" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5b159aceb12908a46702">{"x":{"filter":"none","extensions":["Buttons","FixedColumns","Responsive"],"caption":"<caption>Description of database<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271"],["db_annovar_1000g","db_annovar_1000g","db_annovar_1000g","db_annovar_1000g","db_annovar_1000g","db_annovar_1000g","db_annovar_1000g","db_annovar_1000g","db_annovar_1000g","db_annovar_1000g","db_annovar_1000g","db_annovar_1000g","db_annovar_1000g_sqlite","db_annovar_abraom","db_annovar_avsift","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_annovar_avsnp_sqlite","db_annovar_brvar","db_annovar_brvar","db_annovar_cadd","db_annovar_cadd","db_annovar_cadd","db_annovar_cadd","db_annovar_cadd","db_annovar_cadd","db_annovar_cadd_sqlite","db_annovar_cadd_sqlite","db_annovar_cadd_sqlite","db_annovar_cadd_sqlite","db_annovar_cadd_sqlite","db_annovar_cadd_sqlite","db_annovar_cancer_hotspots","db_annovar_cg","db_annovar_cg","db_annovar_civic_gene_summaries","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar","db_annovar_clinvar_sqlite","db_annovar_clinvar_sqlite","db_annovar_clinvar_sqlite","db_annovar_clinvar_sqlite","db_annovar_cosmic","db_annovar_cosmic","db_annovar_cosmic","db_annovar_cosmic","db_annovar_cosmic","db_annovar_cosmic","db_annovar_cosmic","db_annovar_cosmic_sqlite","db_annovar_cosmic_sqlite","db_annovar_cscd","db_annovar_cscd","db_annovar_cscd","db_annovar_darned_sqlite","db_annovar_dbnsfp_sqlite","db_annovar_dbnsfp_sqlite","db_annovar_dbnsfp_sqlite","db_annovar_dbscsnv","db_annovar_dbscsnv_sqlite","db_annovar_dhs_gene_connectivity","db_annovar_disgenet","db_annovar_disgenet","db_annovar_disgenet","db_annovar_disgenet","db_annovar_disgenet","db_annovar_disgenet","db_annovar_disgenet","db_annovar_docm","db_annovar_eigen","db_annovar_eigen_sqlite","db_annovar_ensgene","db_annovar_epi_genes","db_annovar_esp6500siv2","db_annovar_esp6500siv2","db_annovar_esp6500siv2","db_annovar_exac03","db_annovar_exac03","db_annovar_exac03","db_annovar_exac03_sqlite","db_annovar_exac03_sqlite","db_annovar_exac03_sqlite","db_annovar_fathmm","db_annovar_gdi_score","db_annovar_gerp","db_annovar_gerp","db_annovar_gme","db_annovar_gme_sqlite","db_annovar_gnomad","db_annovar_gnomad","db_annovar_gnomad","db_annovar_gnomad","db_annovar_gnomad_sqlite","db_annovar_gnomad_sqlite","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gtex_eqtl_pairs","db_annovar_gwava","db_annovar_gwava_sqlite","db_annovar_hgnc","db_annovar_hrcr1","db_annovar_hrcr1_sqlite","db_annovar_icgc21","db_annovar_icgc_sqlite","db_annovar_intervar","db_annovar_intervar","db_annovar_intervar_sqlite","db_annovar_intogen","db_annovar_kaviar","db_annovar_knowngene","db_annovar_ljb26_all","db_annovar_lncediting_sqlite","db_annovar_loftool_scores","db_annovar_mcap","db_annovar_mcap","db_annovar_mcap_sqlite","db_annovar_mitimpact","db_annovar_mitimpact","db_annovar_nci60","db_annovar_nci60_sqlite","db_annovar_normal_pool","db_annovar_normal_pool","db_annovar_normal_pool","db_annovar_normal_pool","db_annovar_normal_pool","db_annovar_normal_pool","db_annovar_normal_pool_sqlite","db_annovar_omim_genemap2","db_annovar_popfreq","db_annovar_popfreq","db_annovar_popfreq_sqlite","db_annovar_popfreq_sqlite","db_annovar_radar_sqlite","db_annovar_rddpred_sqlite","db_annovar_rediportal_sqlite","db_annovar_refgene","db_annovar_regsnpintron","db_annovar_revel","db_annovar_revel_sqlite","db_annovar_rvis_esv_score","db_annovar_seeqtl","db_annovar_seeqtl","db_annovar_seeqtl","db_annovar_seeqtl","db_annovar_snp","db_annovar_snp","db_annovar_snp","db_annovar_snp","db_annovar_snp","db_annovar_snp","db_annovar_snp","db_annovar_snp","db_annovar_tall_somatic_genes","db_annovar_tmcsnpdb","db_annovar_varcards","db_annovar_varcards_sqlite","db_ucsc_cytoband","db_ucsc_dnase_clustered","db_ucsc_ensgene","db_ucsc_knowngene","db_ucsc_refgene","db_ucsc_tfbs_clustered"],["1000g2015aug","1000g2014oct","1000g2014sep","1000g2014aug","1000g2012apr","1000g2012feb","1000g2011may","1000g2010nov","1000g2012apr","1000g2010jul","1000g2010","1000g","1000g2015aug","abraom","avsift","avsnp147","avsnp147.common","avsnp144","avsnp142","avsnp138","brvar_v1_core","brvar_v1_extra","cadd","caddgt10","caddgt20","cadd13","cadd13gt10","cadd13gt20","cadd13","cadd13gt10","cadd13gt20","cadd","caddgt10","caddgt20","cancer_hotspots_v2","cg69","cg46","nightly","clinvar_20131105","clinvar_20140211","clinvar_20140303","clinvar_20140702","clinvar_20140902","clinvar_20140929","clinvar_20150330","clinvar_20150629","clinvar_20151201","clinvar_20160302","clinvar_20161128","clinvar_20170130","clinvar_20170501","clinvar_20170905","clinvar_20180603","clinvar_20190305","clinvar_20170130","clinvar_20170501","clinvar_20170905","clinvar_20180603","cosmic70","cosmic68wgs","cosmic68","cosmic67wgs","cosmic67","cosmic65","cosmic64","cosmic70","cosmic81","cscd_cancer_circrna","cscd_common_circrna","cscd_normal_circrna","DARNED","dbnsfp33a","dbnsfp31a_interpro","dbnsfp30a","dbscsnv11","dbscsnv11","dhs_gene_connectivity_2012","disgenet_befree_gene_disease","disgenet_befree_rs_disease","disgenet_curated_gene_disease","disgenet_curated_variant_disease","disgenet_gene_disease","disgenet_pubannotator_variant_disease","disgenet_rs_disease","docm_3.2","eigen","eigen","ensgene","v1.1","esp6500siv2_ea","esp6500siv2_aa","esp6500siv2_all","exac03nontcga","exac03nonpsych","exac03","exac03","exac03nonpsych","exac03nontcga","fathmm","GDI_full_10282015","gerp++elem","gerp++gt2","gme","gme","gnomad_exome","gnomad_genome","gnomad211_exome","gnomad211_genome","gnomad_exome","gnomad_genome","adipose_subcutaneous","adipose_visceral_omentum","adrenal_gland","artery_aorta","artery_coronary","artery_tibial","brain_amygdala","brain_anterior_cingulate_cortex_ba24","brain_caudate_basal_ganglia","brain_cerebellar_hemisphere","brain_cerebellum","brain_cortex","brain_frontal_cortex_ba9","brain_hippocampus","brain_hypothalamus","brain_nucleus_accumbens_basal_ganglia","brain_putamen_basal_ganglia","brain_spinal_cord_cervical_c_1","brain_substantia_nigra","breast_mammary_tissue","cells_ebv_transformed_lymphocytes","cells_transformed_fibroblasts","colon_sigmoid","colon_transverse","esophagus_gastroesophageal_junction","esophagus_mucosa","esophagus_muscularis","heart_atrial_appendage","heart_left_ventricle","liver","lung","minor_salivary_gland","muscle_skeletal","nerve_tibial","ovary","pancreas","pituitary","prostate","skin_not_sun_exposed_suprapubic","skin_sun_exposed_lower_leg","small_intestine_terminal_ileum","spleen","stomach","testis","thyroid","uterus","vagina","whole_blood","adipose_subcutaneous","adipose_visceral_omentum","adrenal_gland","artery_aorta","artery_coronary","artery_tibial","brain_amygdala","brain_anterior_cingulate_cortex_ba24","brain_caudate_basal_ganglia","brain_cerebellar_hemisphere","brain_cerebellum","brain_cortex","brain_frontal_cortex_ba9","brain_hippocampus","brain_hypothalamus","brain_nucleus_accumbens_basal_ganglia","brain_putamen_basal_ganglia","brain_spinal_cord_cervical_c_1","brain_substantia_nigra","breast_mammary_tissue","cells_ebv_transformed_lymphocytes","cells_transformed_fibroblasts","colon_sigmoid","colon_transverse","esophagus_gastroesophageal_junction","esophagus_mucosa","esophagus_muscularis","heart_atrial_appendage","heart_left_ventricle","liver","lung","minor_salivary_gland","muscle_skeletal","nerve_tibial","ovary","pancreas","pituitary","prostate","skin_not_sun_exposed_suprapubic","skin_sun_exposed_lower_leg","small_intestine_terminal_ileum","spleen","stomach","testis","thyroid","uterus","vagina","whole_blood","gwava","gwava","hgnc_coding","hrcr1","hrcr1","icgc21","icgc21","intervar_20170202","intervar_20180118","intervar_20170202","intogen_20180119","kaviar_20150923","knowngene","ljb26_all","lncediting","LoFtool_scores","mcap","mcap13","mcap","mitimpact2","mitimpact24","nci60","nci60","2016sih_wes_ball","2016sih_wes_tall","2016sih_wes_nkt","2016sih_wgs_nkt","2016sih_wgs_dlbcl","2018sih_wgs_nkt_full","2018sih_wgs_nkt_full_sqlite","omim_genemap2","popfreq_max_20150413","popfreq_all_20150413","popfreq_all_20150413","popfreq_max_20150413","RADAR2","rddpred_mes","REDIportal","refgene","regsnpintron","revel","revel","RVIS_ExAC_4KW","seeqtl_qvalue_hapmap3_cis","seeqtl_qvalue_hapmap3_trans","seeqtl_qvalue_myers_cis","seeqtl_qvalue_myers_trans","snp138","snp137","snp135","snp132","snp131","snp130","snp129","snp128","20171206","tmcsnpdb","varcards","varcards","cytoband","V3","ensgene","knowngene","refgene","V3"],["alternative allele frequency data in 1000 Genomes Project","alternative allele frequency data in 1000 Genomes Project","alternative allele frequency data in 1000 Genomes Project","alternative allele frequency data in 1000 Genomes Project","alternative allele frequency data in 1000 Genomes Project","alternative allele frequency data in 1000 Genomes Project","alternative allele frequency data in 1000 Genomes Project","alternative allele frequency data in 1000 Genomes Project","alternative allele frequency data in 1000 Genomes Project","alternative allele frequency data in 1000 Genomes Project","alternative allele frequency data in 1000 Genomes Project","alternative allele frequency data in 1000 Genomes Project","Same with ANNOVAR item","abraom: 2.3 million [Brazilian genomic variants](https://www.ncbi.nlm.nih.gov/pubmed/28332257)","whole-exome SIFT scores for non-synonymous variants (obselete and should not be uesd any more)","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","Integrated 1285 cases B-cell acute lymphoblastic leukemia (B-ALL) from public data portal. Variant calling result, using GATK (Haplotypecaller, UnifiedGenotyper), Lofreq and Varscan2, be merged (bases with baseQ/BAQ &gt;13;minimum coverge&gt;=10; maximum alt allele frequency &gt;= 0.04; alt allel read number &gt;= 3; only the varinats that mutation type is missense, nonsense, frameshift del, frameshift del, nonframeshift del, nonframeshift del, stoploss, nonsense and unknown be included)","Complementary set of core, such as Synonymous mutation, intron, intergenic, UTR3 and UTR5, .etc","CADD score; 13-version 1.3; gt10-score&gt;10; gt20-score&gt;20","CADD score; 13-version 1.3; gt10-score&gt;10; gt20-score&gt;20","CADD score; 13-version 1.3; gt10-score&gt;10; gt20-score&gt;20","CADD score; 13-version 1.3; gt10-score&gt;10; gt20-score&gt;20","CADD score; 13-version 1.3; gt10-score&gt;10; gt20-score&gt;20","CADD score; 13-version 1.3; gt10-score&gt;10; gt20-score&gt;20","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","See http://cancerhotspots.org/#/home Hotspot Results V2 sheet1","allele frequency in 69 human subjects sequenced by Complete Genomics","alternative allele frequency in 46 unrelated human subjects sequenced by Complete Genomics","Same with ANNOVAR item","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","CLINVAR database with Variant Clinical Significance (unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response, histocompatibility, other) and Variant disease name","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","COSMIC database version","COSMIC database version on WGS data","COSMIC database version","COSMIC database version on WGS data","COSMIC database version","COSMIC database version","COSMIC database version","Same with ANNOVAR item","Same with ANNOVAR item","Download from http://gb.whu.edu.cn/CSCD/, Cancer specific circRNA database.","Download from http://gb.whu.edu.cn/CSCD/, Cancer specific circRNA database.","Download from http://gb.whu.edu.cn/CSCD/, Cancer specific circRNA database.","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","dbscSNV version 1.1 for splice site prediction by AdaBoost and Random Forest","dbscSNV version 1.1 for splice site prediction by AdaBoost and Random Forest","Genomic coordinates of all promoter DHSs and distal, non-promoter DHSs within +-500 kb correlated with them at threshold 0.7.","See http://www.disgenet.org/web/DisGeNET/menu/downloads","See http://www.disgenet.org/web/DisGeNET/menu/downloads","See http://www.disgenet.org/web/DisGeNET/menu/downloads","See http://www.disgenet.org/web/DisGeNET/menu/downloads","See http://www.disgenet.org/web/DisGeNET/menu/downloads","See http://www.disgenet.org/web/DisGeNET/menu/downloads","See http://www.disgenet.org/web/DisGeNET/menu/downloads","DoCM, the Database of Curated Mutations, is a highly curated database of known, disease-causing mutations that provides easily explorable variant lists with direct links to source citations for easy verification. http://docm.genome.wustl.edu/about","whole-genome Eigen scores, see ref (http://www.ncbi.nlm.nih.gov/pubmed/26727659)","Same with ANNOVAR item","FASTA sequences for all annotated transcripts in ENSEMBL Gene","Epigenetic genes defined in 'The landscape of somatic mutations in epigenetic regulators across 1,000 paediatric cancer genomes', doi:10.1038/ncomms4630","alternative allele frequency in All subjects in the NHLBI-ESP project with 6500 exomes, including the indel calls and the chrY calls. This is lifted over from hg19 by myself.","alternative allele frequency in European American subjects in the NHLBI-ESP project with 6500 exomes, including the indel calls and the chrY calls. This is lifted over from hg19 by myself","alternative allele frequency in African American subjects in the NHLBI-ESP project with 6500 exomes, including the indel calls and the chrY calls. This is lifted over from hg19 by myself.","ExAC on non-TCGA samples (updated header)","ExAC on non-Psychiatric disease samples (updated header)","ExAC 65000 exome allele frequency data for ALL, AFR (African), AMR (Admixed American), EAS (East Asian), FIN (Finnish), NFE (Non-finnish European), OTH (other), SAS (South Asian)). version 0.3. Left normalization done.","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","whole-genome FATHMM_coding and FATHMM_noncoding scores (noncoding and coding scores in the 2015 version was reversed)","GDI score: the gene damage index (GDI) is describing the accumulated mutational damage for each human gene in the general population, and shows that highly mutated/damaged genes are unlikely to be disease-causing and yet they generate a big proportion of false positive variants harbored in such genes. Therefore removing high GDI genes is a very effective way to remove confidently false positives from WES/WGS data. More details were given in [this paper](http://www.pnas.org/content/early/2015/10/14/1518646112.abstract). The data set includes general damage prediction (low/medium/high) for different disease type (all, Mendelian, cancer, and PID) and can be downloaded from here.","Same with ANNOVAR item","Same with ANNOVAR item","Great Middle East allele frequency including NWA (northwest Africa), NEA (northeast Africa), AP (Arabian peninsula), Israel, SD (Syrian desert), TP (Turkish peninsula) and CA (Central Asia)","Same with ANNOVAR item",null,"gnomAD genome collection","gnomAD exome collection (v2.1.1), with \\\"AF AF_popmax AF_male AF_female AF_raw AF_afr AF_sas AF_amr AF_eas AF_nfe AF_fin AF_asj AF_oth non_topmed_AF_popmax non_neuro_AF_popmax non_cancer_AF_popmax controls_AF_popmax\\\" header","same above all","Same with ANNOVAR item","Same with ANNOVAR item","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","eQTLs of GTEx significant variant gene paris.","whole genome GWAVA_region_score GWAVA_tss_score GWAVA_unmatched_score, see ref (http://www.nature.com/nmeth/journal/v11/n3/abs/nmeth.2832.html)","Same with ANNOVAR item","HGNC is responsible for approving unique symbols and names for human loci, including protein coding genes, ncRNA genes and pseudogenes, to allow unambiguous scientific communication. hg19_hgnc_coding is the ftp://ftp.ebi.ac.uk/pub/databases/genenames/new/tsv/hgnc_complete_set.txt","40 million variants from 32K samples in haplotype reference consortium","Same with ANNOVAR item","International Cancer Genome Consortium version 21","Same with ANNOVAR item","InterVar: clinical interpretation of missense variants","InterVar: clinical interpretation of missense variants","Same with ANNOVAR item","Merged intogen table intogen_mutations_catalog [Driver or passenger mutations]. https://www.intogen.org/downloads","170 million Known VARiants from 13K genomes and 64K exomes in 34 projects","FASTA sequences for all annotated transcripts in UCSC Known Gene","whole-exome SIFT, PolyPhen2 HDIV, PolyPhen2 HVAR, LRT, MutationTaster, MutationAssessor, FATHMM, MetaSVM, MetaLR, VEST, CADD, GERP++, PhyloP and SiPhy scores from dbNSFP version 2.6","Same with ANNOVAR item","LoFtool score: gene loss-of-function score percentiles. The smaller the percentile, the most intolerant is the gene to functional variation. The file can be downloaded [here](http://clark.compbio.iupui.edu/regsnp_intron_web/). Manuscript in preparation (please contact Dr. Joao Fadista - joao.fadista@med.lu.se). The authors would like to thank the Exome Aggregation Consortium and the groups that provided exome variant data for comparison. A full list of contributing groups can be found at http://exac.broadinstitute.org/about.","M-CAP scores for non-synonymous variants; mcap13:M-CAP scores v1.3, 20181203","M-CAP scores for non-synonymous variants; mcap13:M-CAP scores v1.3, 20181203","Same with ANNOVAR item","pathogenicity predictions of human mitochondrial missense variants (http://www.ncbi.nlm.nih.gov/m/pubmed/25516408/)","pathogenicity predictions of human mitochondrial missense variants (http://www.ncbi.nlm.nih.gov/m/pubmed/25516408/)","NCI-60 human tumor cell line panel exome sequencing allele frequency data","Same with ANNOVAR item","Variants called from normal control samples (SIH)","Variants called from normal control samples (SIH)","Variants called from normal control samples (SIH)","Variants called from normal control samples (SIH)","Variants called from normal control samples (SIH)","Variants called from normal control samples (SIH)","Variants called from normal control samples (SIH), big files","omim genemap2 txt file, download from http://www.omim.org/","A database containing the maximum allele frequency from 1000G, ESP6500, ExAC and CG46","A database containing all allele frequency from 1000G, ESP6500, ExAC and CG46","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","Same with ANNOVAR item","FASTA sequences for all annotated transcripts in RefSeq Gene","prioritize the disease-causing probability of intronic SNVs","REVEL scores for non-synonymous variants","Same with ANNOVAR item","RVIS-ESV score: RVIS score measures genetic intolerance of genes to functional mutations, as described in [Petrovski et al.](http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1003709) Original RVIS was constructed based on patterns of standing variation in 6503 samples. The authors have recently constructed scores based on the ~61,000 samples from ExAC. There is high correlation, but more resolution for many genes. The ExAC cohort implementation is what we consider RVIS (v2). It can be downloaded here.","eQTLs of meta-analysis (consensus eQTLs) from HapMap human lymphoblastoid cell lines; eQTLs of human brain from Myers et al, see http://www.bios.unc.edu/research/genomic_software/seeQTL/data_source","eQTLs of meta-analysis (consensus eQTLs) from HapMap human lymphoblastoid cell lines; eQTLs of human brain from Myers et al, see http://www.bios.unc.edu/research/genomic_software/seeQTL/data_source","eQTLs of meta-analysis (consensus eQTLs) from HapMap human lymphoblastoid cell lines; eQTLs of human brain from Myers et al, see http://www.bios.unc.edu/research/genomic_software/seeQTL/data_source","eQTLs of meta-analysis (consensus eQTLs) from HapMap human lymphoblastoid cell lines; eQTLs of human brain from Myers et al, see http://www.bios.unc.edu/research/genomic_software/seeQTL/data_source","dbSNP with ANNOVAR index files","dbSNP with ANNOVAR index files","dbSNP with ANNOVAR index files","dbSNP with ANNOVAR index files","dbSNP with ANNOVAR index files","dbSNP with ANNOVAR index files","dbSNP with ANNOVAR index files","dbSNP with ANNOVAR index files","Collected from recently published papers, 1) Recurrent SPI1 (PU.1) fusions in high-risk pediatric T cell acute lymphoblastic leukemia 2) The genomic landscape of pediatric and young adult T-lineage acute lymphoblastic leukemia 3) Identification of fusion genes and characterization of transcriptome features in T-cell acute lymphoblastic leukemia","TMC-SNPDB: SNP database from whole exome data of 62 normal samples derived from cancer patients of Indian origin, representing 114, 309 unique germline variants. Read the manuscript [here](http://database.oxfordjournals.org/content/2016/baw104.full). It is useful for exome sequencing studies on Indian populations and can be downloaded from here.","Variants data download from http://varcards.biols.ac.cn, Varcards is an integrated genetic and clinical database for coding variants in the human genome. doi:10.1093/nar/gkx1039","Same with ANNOVAR item","UCSC goldenPath/{{buildver}}/database/cytoBand.txt.gz","UCSC DNaseI Hypersensitivity Clusters in 125 cell types from ENCODE","UCSC goldenPath/{{buildver}}/database/ensGene.txt.gz","UCSC goldenPath/{{buildver}}/database/knownGene.txt.gz and kgXref.txt.gz","UCSC goldenPath/{{buildver}}/database/refGene.txt.gz","UCSC Transcription Factor ChIP-seq (161 factors) from ENCODE with Factorbook Motifs"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Database<\/th>\n      <th>Version<\/th>\n      <th>Description<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}],"responsive":true}},"evals":[],"jsHooks":[]}</script><p>Besides, you can use the function <code>download.database</code> with parameter <code>show.all.buildvers = TRUE</code> get all available buildver.</p>
<html><body><pre class="r"><span class="no">x</span> <span class="kw">&lt;-</span> <span class="kw pkg">anor</span><span class="kw ns">::</span><span class="fu">download.database</span>(<span class="kw">download.name</span> <span class="kw">=</span> <span class="st">"db_annovar_cosmic"</span>, <span class="kw">show.all.buildvers</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">x</span>)
<span class="co">#&gt;          db_annovar_cosmic</span>
<span class="co">#&gt; cosmic70        hg38, hg19</span>
<span class="co">#&gt; other                 hg19</span></pre></body></html>
</div>
<div id="gene-and-clincal-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-and-clincal-annotation" class="anchor"></a>Gene and clincal annotation</h2>
<p>Gene annotation databases contain the gene classification, gene function and phenotype correlation, such as <a href="https://www.genenames.org/">HGNC</a>, <a href="http://www.omim.org/">OMIM</a> <a href="http://docm.genome.wustl.edu/">DoCM</a>, <a href="https://civic.genome.wustl.edu/">CIVic</a>, <a href="http://www.disgenet.org/web/DisGeNET/menu/browser?1">DisGeNET</a>, <a href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a>, and Gene Ontology (GO), .etc.</p>
</div>
<div id="variant-effect-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#variant-effect-prediction" class="anchor"></a>Variant effect prediction</h2>
<p>Variant effect prediction databases contain the various databases generated by the algorithms for prediction of variants effect on protein or RNA structural, such as <a href="http://sift.jcvi.org/">SIFT</a>, <a href="http://genetics.bwh.harvard.edu/pph2/">PolyPhen2</a>, <a href="http://provean.jcvi.org">PROVEAN</a>, <a href="http://www.mutationtaster.org/">MutationTaster</a>, <a href="http://mutationassessor.org/r3/">MutationAssessor</a>, <a href="http://fathmm.biocompute.org.uk/">FATHMM</a>, .etc.</p>
</div>
<div id="population-allele-frequency" class="section level2">
<h2 class="hasAnchor">
<a href="#population-allele-frequency" class="anchor"></a>Population allele frequency</h2>
<p>Population allele frequency databases contain the databases based on the population cohort genome sequencing data (mainly include whole genome sequencing and whole exome sequencing), such as <a href="https://www.ncbi.nlm.nih.gov/projects/faspftp/1000genomes/">1000 Genome Project</a>, <a href="https://esp.gs.washington.edu/drupal/">NHLBI GO Exome Sequencing Project (ESP)</a>, <a href="http://gnomad.broadinstitute.org/">gnomAD</a> and <a href="http://exac.broadinstitute.org/">ExAC</a>, .etc.</p>
</div>
<div id="cancer-somatic-mutation" class="section level2">
<h2 class="hasAnchor">
<a href="#cancer-somatic-mutation" class="anchor"></a>Cancer somatic mutation</h2>
<p>Cancer somatic mutation databases generated by the cancer patients case-control paired genomic sequence data, such as <a href="http://cancer.sanger.ac.uk/cosmic">COSMIC</a>, Cancer Hotspots, <a href="https://www.intogen.org/downloads">intogen</a> and <a href="https://www.cancergenomeinterpreter.org/biomarkers">Cancer Biomarkers database</a>, .etc.</p>
</div>
<div id="rna-seq-variants" class="section level2">
<h2 class="hasAnchor">
<a href="#rna-seq-variants" class="anchor"></a>RNA-seq variants</h2>
<p>RNA-seq variants databases contributed by variants called from RNA-seq including expressed allele and RNA-editing. anor built an RNA-seq variants database, BRVar, based on 1285 cases B-cell lymphoblastic leukemia (B-ALL) patients RNA-seq data (Four different variants detection method be applied).</p>
</div>
<div id="expression-quantitative-trait-locus-eqtl" class="section level2">
<h2 class="hasAnchor">
<a href="#expression-quantitative-trait-locus-eqtl" class="anchor"></a>Expression quantitative trait locus (eQTL)</h2>
<p>eQTL databases contain the candidate locus of genome that have an candidate impact on gene expression level, such as <a href="https://www.gtexportal.org/home/">Genotype-Tissue Expression (GTEx)</a> QTL, <a href="http://www.bios.unc.edu/research/genomic_software/seeQTL/">seeQTL</a> and <a href="http://bioinfo.life.hust.edu.cn/PancanQTL/">PancanQTL</a>, .etc.</p>
</div>
<div id="non-coding-rna" class="section level2">
<h2 class="hasAnchor">
<a href="#non-coding-rna" class="anchor"></a>Non-coding RNA</h2>
<p>Non-coding RNA databases contain the candidate biomarkers or non-coding RNA targeted transcriptional regulation region, such as <a href="http://gb.whu.edu.cn/CSCD/">Cancer-Specific CirRNA Database</a> and (LNCediting)[<a href="http://bioinfo.life.hust.edu.cn/LNCediting/">http://bioinfo.life.hust.edu.cn/LNCediting/</a>], .etc.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="life">
  <p>anor was developed by <strong><a href="http://life2cloud.com">Jianfeng</a></strong>. </p>
</div>

<div class="author">
  <p>Developed by Jianfeng Li.</p>
  <p>Site built by <a href="http://pkgdown.r-lib.org">pkgdown</a>.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
