<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to anor • anor</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="../life.css" rel="stylesheet">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- life --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a class="navbar-brand" href="../index.html">anor</a>
        <small class="life">developed by <a href="http://life2cloud.com">Jianfeng</a></small>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li>
  <a href="../articles/introduction_to_anor.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/database_in_anor.html">Database in anor</a>
    </li>
    <li>
      <a href="../articles/shiny_of_anor.html">Shiny app of anor</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
        <li>
  <a href="https://github.com/JhuangLab/anor">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="introduction_to_anor_files/htmlwidgets-1.5.1/htmlwidgets.js"></script><script src="introduction_to_anor_files/jquery-1.12.4/jquery.min.js"></script><link href="introduction_to_anor_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="introduction_to_anor_files/datatables-binding-0.13/datatables.js"></script><link href="introduction_to_anor_files/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="introduction_to_anor_files/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="introduction_to_anor_files/dt-core-1.10.20/js/jquery.dataTables.min.js"></script><link href="introduction_to_anor_files/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet">
<script src="introduction_to_anor_files/crosstalk-1.1.0.1/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to anor</h1>
                        <h4 class="author">Jianfeng Li</h4>
            
            <h4 class="date">2020-04-20</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/JhuangLab/anor/blob/master/vignettes/introduction_to_anor.Rmd"><code>vignettes/introduction_to_anor.Rmd</code></a></small>
      <div class="hidden name"><code>introduction_to_anor.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Interpretation of genetic variation data is a crucial step to understand the relationship between gene sequence changes and biological function. There are several annotation tools, such as <a href="http://annovar.openbioinformatics.org/">ANNOVAR</a>, <a href="https://asia.ensembl.org/info/docs/tools/vep/index.html">VEP</a>, <a href="https://github.com/brentp/vcfanno">vcfanno</a>, have been developed. These tools make gene variation data annotation more convenient and faster than before. However, because different annotation tools have their own methods of use and design architecture, this increases the difficulty for bioinformatics beginner to utilize these tools. In addition, many of existing database resources and annotation scripts have not been well integrated and shared.</p>
<p>So, it is worth us to develop an integrated annotation system that not only includes the integration of different annotation tools but also integrate the relevant database resources. Here, we present an integrated annotation R package ‘<a href="https://github.com/JhuangLab/anor">anor</a>’ to do this. It provides a series R functions to integrate external annotation tools and annotation databases.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install anor, you need to install <a href="https://cran.r-project.org/">R</a> interpreter (Supported Linux, MAC, and Windows). This package has been uploaded on The Comprehensive R Archive Network (CRAN, <a href="https://cran.r-project.org">https://cran.r-project.org</a>). You can use the command to install anor package easily:</p>
<html><body><pre class="r"><span class="co"># setRepositories ind 1 is CRAN, 2 is Bioconductor</span>
<span class="fu"><a href="https://rdrr.io/r/utils/setRepositories.html">setRepositories</a></span>(<span class="kw">ind</span><span class="kw">=</span><span class="fl">1</span>:<span class="fl">2</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">'anor'</span>)</pre></body></html>
<p>If you want to use the latest development version, you need to use <a href="https://github.com/hadley/devtools">devtools</a> <code>install_github</code> function.</p>
<html><body><pre class="r"><span class="co"># Install the cutting edge development version from GitHub:</span>
<span class="co"># install.packages("devtools")</span>
<span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"JhuangLab/anor"</span>, <span class="kw">ref</span> <span class="kw">=</span> <span class="st">"develop"</span>)</pre></body></html>
<p>anor can also be installed using the source code archive (<code>R CMD INSTALL</code>). In this situation, you need to manually handle dependencies on many packages.</p>
<p><strong>Tips</strong>: When the <a href="https://github.com/r-dbi/RMySQL">RMySQL</a> or <a href="https://github.com/r-dbi/RSQLite">RSQLite</a> package can not directly installed by R, <a href="https://conda.io/docs/">conda</a> is an optional solution: <code>conda install -c r r-rmysql r-rsqlite</code>. Or you need root permissions to install the corresponding system dependency.</p>
</div>
<div id="download-database-resource" class="section level2">
<h2 class="hasAnchor">
<a href="#download-database-resource" class="anchor"></a>Download database resource</h2>
<p>To reduce the procedure of download database and other material, anor provides single function <code>download.database</code> to download various annotation databases for <a href="http://annovar.openbioinformatics.org/en/latest/">ANNOVAR</a>, <a href="http://www.bioconductor.org/packages/devel/bioc/html/AnnotationDbi.html">AnnotationDbi</a>, <a href="https://github.com/brentp/vcfanno">vcfanno</a>, <a href="http://asia.ensembl.org/info/docs/tools/vep/index.html">vep</a>, <a href="http://portals.broadinstitute.org/oncotator/">oncotator</a> and <a href="https://github.com/ryanlayer/giggle">giggle</a>. Moreover, you can share the database configuration for others to download your database in protected (license code required) or unprotected mode.</p>
<html><body><pre class="r"><span class="co"># Library needed R package</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BioInstaller</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">anor</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">data.table</span>)

<span class="co"># Show all anor supported database</span>
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="kw">show.all.names</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html>
<div id="htmlwidget-ee9f0c33197ad48bdbe7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ee9f0c33197ad48bdbe7">{"x":{"filter":"none","caption":"<caption>Annotation database supported by anor (n=86)<\/caption>","data":[["db_annovar_1000g","db_annovar_1000g_sqlite","db_annovar_abraom","db_annovar_avsift","db_annovar_avsnp","db_annovar_avsnp_sqlite","db_annovar_brvar","db_annovar_cadd","db_annovar_cadd_sqlite","db_annovar_cancer_hotspots","db_annovar_cg","db_annovar_civic_gene_summaries","db_annovar_clinvar","db_annovar_clinvar_sqlite","db_annovar_cosmic","db_annovar_cosmic_sqlite","db_annovar_cscd","db_annovar_darned_sqlite","db_annovar_dbnsfp","db_annovar_dbnsfp_sqlite","db_annovar_dbscsnv","db_annovar_dbscsnv_sqlite","db_annovar_dhs_gene_connectivity","db_annovar_disgenet","db_annovar_docm","db_annovar_eigen","db_annovar_eigen_sqlite","db_annovar_ensgene","db_annovar_epi_genes"],["db_annovar_esp6500siv2","db_annovar_exac03","db_annovar_exac03_sqlite","db_annovar_fathmm","db_annovar_gdi_score","db_annovar_gerp","db_annovar_gme","db_annovar_gme_sqlite","db_annovar_gnomad","db_annovar_gnomad_sqlite","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_pairs","db_annovar_gwava","db_annovar_gwava_sqlite","db_annovar_hgnc","db_annovar_hrcr1","db_annovar_hrcr1_sqlite","db_annovar_icgc21","db_annovar_icgc_sqlite","db_annovar_intervar","db_annovar_intervar_sqlite","db_annovar_intogen","db_annovar_kaviar","db_annovar_knowngene","db_annovar_ljb26_all","db_annovar_lncediting_sqlite","db_annovar_loftool_scores","db_annovar_mcap","db_annovar_mcap_sqlite"],["db_annovar_mitimpact","db_annovar_nci60","db_annovar_nci60_sqlite","db_annovar_normal_pool","db_annovar_normal_pool_sqlite","db_annovar_omim_genemap2","db_annovar_popfreq","db_annovar_popfreq_sqlite","db_annovar_radar_sqlite","db_annovar_rddpred_sqlite","db_annovar_rediportal_sqlite","db_annovar_refgene","db_annovar_regsnpintron","db_annovar_revel","db_annovar_revel_sqlite","db_annovar_rvis_esv_score","db_annovar_seeqtl","db_annovar_snp","db_annovar_tall_somatic_genes","db_annovar_tmcsnpdb","db_annovar_varcards","db_annovar_varcards_sqlite","db_ucsc_cytoband","db_ucsc_dnase_clustered","db_ucsc_ensgene","db_ucsc_knowngene","db_ucsc_refgene","db_ucsc_tfbs_clustered","db_annovar_1000g"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>V1<\/th>\n      <th>V2<\/th>\n      <th>V3<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><html><body><pre class="r"><span class="co"># Show all supported version of database (e.g. db_annovar_avsnp)</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="kw">download.name</span> <span class="kw">=</span> <span class="st">"db_annovar_avsnp"</span>, <span class="kw">show.all.version</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="co">#&gt; [1] "avsnp150" "avsnp147" "avsnp144" "avsnp142" "avsnp138"</span>

<span class="co"># Show all supprted buildver of specific version database</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="kw">download.name</span> <span class="kw">=</span> <span class="st">"db_annovar_avsnp"</span>, <span class="kw">version</span> <span class="kw">=</span> <span class="st">"avsnp147"</span>, <span class="kw">show.all.buildvers</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="co">#&gt;          db_annovar_avsnp</span>
<span class="co">#&gt; avsnp138 "hg19"          </span>
<span class="co">#&gt; avsnp142 Character,2     </span>
<span class="co">#&gt; avsnp144 Character,2     </span>
<span class="co">#&gt; avsnp147 Character,2     </span>
<span class="co">#&gt; avsnp150 Character,2</span>

<span class="co"># To reduce the download time, we use the local demo configuratin file to download demo file</span>
<span class="no">demo.cfg</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"demo/demo.cfg"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"anor"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="st">"download_demo"</span>, <span class="kw">show.all.versions</span> <span class="kw">=</span> <span class="no">T</span>, <span class="kw">download.cfg</span> <span class="kw">=</span> <span class="no">demo.cfg</span>)
<span class="co">#&gt; [1] "demo"</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/download.database.html">download.database</a></span>(<span class="st">"download_demo"</span>, <span class="st">"demo"</span>, <span class="kw">buildver</span> <span class="kw">=</span> <span class="st">"GRCh37"</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span>(<span class="st">"%s/databases/"</span>,
  <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>()), <span class="kw">download.cfg</span> <span class="kw">=</span> <span class="no">demo.cfg</span>)
<span class="co">#&gt; [1] FALSE</span>

<span class="co"># Protected mode</span>
<span class="co"># download.database("your_db_name", license = "license_code")</span>

<span class="co"># If you want to download other resource in BioInstaller,</span>
<span class="co"># you can use function `install.bioinfo`</span>
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BioInstaller/man/install.bioinfo.html">install.bioinfo</a></span>(<span class="kw">show.all.names</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html>
<div id="htmlwidget-5a7b3f676d771dfcfeaa" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5a7b3f676d771dfcfeaa">{"x":{"filter":"none","caption":"<caption>Items supported by BioInstaller (n=383)<\/caption>","data":[["abyss","arnapipe","asap","backspin","bamtools","bamutil","bcftools","bearscc","bedtools","bowtie","bowtie2","bpipe","breakdancer","brie","bwa","cesam","chia-pet2","chromtime","chronqc","cnvkit","cnvnator","dart","delly","dreg","facets","fastp","fastq_tools","fastx_toolkit","freebayes","freec","fsclvm","genomedisco","giggle","github_demo","hisat2","htseq","igraph","isop","jaffa","jvarkit","libgtextutils","lofreq","macs","manta","marvel","mdseq","mimosca","mrnn","mscentipede","multiqc","oases","olego","oncodriveclust","oncotator","outrigger","picard","picky","pindel","piq","pxz","raceid","radia","rca","resm","rhat","rop","rum","samtools_old","sclvm","scnorm","seqtk","sequenza","seurat","singlesplice","sleuth","snakemake","snakemake_dna_gatk_flow","somaticsniper","spack","sparsehash","speedseq","star","strawberry","svaba","taxmaps","tmap","tophat2","tracer","trimgalore","trinityrnaseq","varscan2","vcflib","vcftools","vep","virchip","zifa","absolute","annovar","armadillo","atlas2","autochrom3d","bcl2fastq","beagle","blast","blat","bzip2","cesa","chromhmm","cnvnator_samtools","contest","cromwell","curl","demo_2","easeq","edena","effusion","ensemble_grch37_reffa","ensemble_grch38_reffa","fastqc","fatotwobit","fusioncatcher","fusioncatcher_reffa","gatk","gatk4","gatk4_jhuang_mirror","gatk_bundle","gmap","gridss"],["hapseg","hgtid","hisat2_reffa","htslib","igv","imagej","interproscan","jaffa_reffa","liftover","lzo","lzop","maorm","mapsplice2","marina","meerkat","miniconda2","miniconda3","mutect","mutsig","mutsig_dependence","mutsig_reffa","nextflow","ngs_qc_toolkit","novoalign","paradigm","pcre","pigz","prada","prinseq","quest","r","reditools","rmats","rmats_reffa","root","samstat","samtools","snpeff","solexaqa","sqlite","sratools","srnanalyzer","ssaha2","strelka","subread","svtoolkit","tvc","ucsc_reffa","ucsc_utils","useq","vadir","vcfanno","velvet","xz","zlib","db_appris","db_atcircdb","db_awesome","db_biosystems","db_cancer_hotspots","db_cancersplicingqtl","db_cellmarker","db_cgi","db_circbase","db_circnet","db_circrnadb","db_civic","db_consensuspathdb","db_cscd","db_dbsno","db_dcdb","db_denovo_db","db_dgidb","db_differentialnet","db_diseaseenhancer","db_disgenet","db_docm","db_drugbank","db_ecodrug","db_eggnog","db_ewasdb","db_exorbase","db_expression_atlas","db_exsnp","db_fantom_cage_peaks","db_fantom_co_expression_clusters","db_fantom_enhancers","db_fantom_motifs","db_fantom_ontology","db_fantom_tss_classifier","db_funcoup","db_fusiongdb","db_gtex","db_hgnc","db_hmdb","db_hpdi","db_hpo","db_inbiomap","db_instruct","db_interpro","db_intogen","db_islandviewer","db_lnc2cancer","db_lncediting","db_lncipedia","db_lncrnadisease","db_medreaders","db_mircancer","db_mirdb","db_mirnest","db_mirtarbase","db_mndr","db_msdd","db_msigdb","db_omim_open","db_omim_private","db_oncokb","db_oncomirdb","db_oncotator","db_pancanqtl","db_phosphonetworks","db_pmkb","db_proteinatlas","db_rbp_var","db_rbpdb","db_rddpred","db_redoxdb","db_remap"],["db_remap2","db_rsnp3","db_rvarbase","db_sedb","db_seecancer","db_seeqtl","db_sm2mir","db_snipa3","db_srnanalyzer","db_superdrug2","db_tumorfusions","db_varcards","db_annovar_1000g","db_annovar_1000g_sqlite","db_annovar_abraom","db_annovar_avsift","db_annovar_avsnp","db_annovar_avsnp_sqlite","db_annovar_brvar","db_annovar_cadd","db_annovar_cadd_sqlite","db_annovar_cancer_hotspots","db_annovar_cg","db_annovar_civic_gene_summaries","db_annovar_clinvar","db_annovar_clinvar_sqlite","db_annovar_cosmic","db_annovar_cosmic_sqlite","db_annovar_cscd","db_annovar_darned_sqlite","db_annovar_dbnsfp","db_annovar_dbnsfp_sqlite","db_annovar_dbscsnv","db_annovar_dbscsnv_sqlite","db_annovar_dhs_gene_connectivity","db_annovar_disgenet","db_annovar_docm","db_annovar_eigen","db_annovar_eigen_sqlite","db_annovar_ensgene","db_annovar_epi_genes","db_annovar_esp6500siv2","db_annovar_exac03","db_annovar_exac03_sqlite","db_annovar_fathmm","db_annovar_gdi_score","db_annovar_gerp","db_annovar_gme","db_annovar_gme_sqlite","db_annovar_gnomad","db_annovar_gnomad_sqlite","db_annovar_gtex_eqtl_egenes","db_annovar_gtex_eqtl_pairs","db_annovar_gwava","db_annovar_gwava_sqlite","db_annovar_hgnc","db_annovar_hrcr1","db_annovar_hrcr1_sqlite","db_annovar_icgc21","db_annovar_icgc_sqlite","db_annovar_intervar","db_annovar_intervar_sqlite","db_annovar_intogen","db_annovar_kaviar","db_annovar_knowngene","db_annovar_ljb26_all","db_annovar_lncediting_sqlite","db_annovar_loftool_scores","db_annovar_mcap","db_annovar_mcap_sqlite","db_annovar_mitimpact","db_annovar_nci60","db_annovar_nci60_sqlite","db_annovar_normal_pool","db_annovar_normal_pool_sqlite","db_annovar_omim_genemap2","db_annovar_popfreq","db_annovar_popfreq_sqlite","db_annovar_radar_sqlite","db_annovar_rddpred_sqlite","db_annovar_rediportal_sqlite","db_annovar_refgene","db_annovar_regsnpintron","db_annovar_revel","db_annovar_revel_sqlite","db_annovar_rvis_esv_score","db_annovar_seeqtl","db_annovar_snp","db_annovar_tall_somatic_genes","db_annovar_tmcsnpdb","db_annovar_varcards","db_annovar_varcards_sqlite","db_ucsc_cytoband","db_ucsc_dnase_clustered","db_ucsc_ensgene","db_ucsc_knowngene","db_ucsc_refgene","db_ucsc_tfbs_clustered","db_blast_env_nr","db_blast_est_human","db_blast_est_mouse","db_blast_est_others","db_blast_gss","db_blast_htgs","db_blast_human_genomic","db_blast_landmark","db_blast_mouse_genomic","db_blast_nr","db_blast_nt","db_blast_other_genomic","db_blast_pataa","db_blast_patnt","db_blast_pdbaa","db_blast_pdbnt","db_blast_ref_prok_rep_genomes","db_blast_ref_viroids_rep_genomes","db_blast_ref_viruses_rep_genomes","db_blast_refseq_genomic","db_blast_refseq_protein","db_blast_refseq_rna","db_blast_refseqgene","db_blast_sts","db_blast_swissprot","db_blast_taxdb","db_blast_tsa_nr","db_blast_tsa_nt","db_blast_vector","abyss"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>V1<\/th>\n      <th>V2<\/th>\n      <th>V3<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="basic-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#basic-annotation" class="anchor"></a>Basic annotation</h2>
<p><code>anno.name</code> is the key of shortcut annotation, you can get the annotation name through function <code><a href="https://rdrr.io/pkg/anor/man/get.annotation.names.html">get.annotation.names()</a></code>, or you can search the <a href="https://life2cloud.com/tools/anor/articles/database_in_anor.html#overview-of-supported-database.html">vignette</a> to get the <code>annotation name</code></p>
<p>Besides, <code>download.name</code> is also required for you to download the related database resource connected with <code>anno.name</code>. Now, we provides the function <code>get.download.name</code> to get it.</p>
<html><body><pre class="r"><span class="co"># Get all supprted anno.name in anor</span>
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/get.annotation.names.html">get.annotation.names</a></span>()</pre></body></html>
<div id="htmlwidget-42be868dd18d2882bf3f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-42be868dd18d2882bf3f">{"x":{"filter":"none","caption":"<caption>Annotation database supported by anor (n=215)<\/caption>","data":[["1000g2015aug_all","1000g2015aug_afr","1000g2015aug_amr","1000g2015aug_eas","1000g2015aug_eur","1000g2015aug_sas","perl_annovar_refGene","perl_annovar_ensGene","perl_annovar_knownGene","perl_annovar_cytoBand","perl_annovar_genomicSuperDups","perl_annovar_clinvar_20170905","perl_annovar_esp6500siv2_all","perl_annovar_1000g2015aug_all","perl_annovar_1000g2015aug_afr","perl_annovar_1000g2015aug_eas","perl_annovar_1000g2015aug_eur","perl_annovar_avsnp142","perl_annovar_avsnp144","perl_annovar_avsnp147","perl_annovar_avsnp150","perl_annovar_dbnsfp33a","perl_annovar_dbnsfp35a","perl_annovar_dbnsfp35c","perl_annovar_cosmic70","perl_annovar_cosmic81","perl_annovar_cosmic82","perl_annovar_cosmic83_coding","perl_annovar_cosmic84_coding","perl_annovar_cosmic85_coding","perl_annovar_cosmic86_coding","perl_annovar_cosmic87_coding","perl_annovar_cosmic88_coding","perl_annovar_eigen","perl_annovar_gwava","perl_annovar_cadd13","perl_annovar_cadd13gt10","perl_annovar_cadd13gt20","perl_annovar_regsnpintron","perl_annovar_mitimpact24","perl_annovar_popfreq_all_20150413","perl_annovar_popfreq_max_20150413","perl_annovar_icgc21","perl_annovar_nci60","perl_annovar_gme","perl_annovar_hrcr1","perl_annovar_kaviar_20150923","perl_annovar_gnomad_genome","perl_annovar_gnomad_exome","perl_annovar_exac03nonpsych","perl_annovar_exac03nontcga","perl_annovar_exac03","perl_annovar_cg69","perl_annovar_cg46","perl_annovar_intervar_20170202","perl_annovar_intervar_20180118","perl_annovar_dbscsnv11","perl_annovar_merge","avsnp138","avsnp142","avsnp144","avsnp147","avsnp150","bioc_org_hs_eg","bioc_gene2","bioc_gene2alias","bioc_gene2ensembl_gene_id","bioc_gene2entrez_gene_id","bioc_gene2gene_full_name","brvar_v1_core","brvar_v1_extra","cadd"],["caddgt10","caddgt20","cadd13","cadd13gt10","cadd13gt20","cancer_hotspots_v2","nightly_civic_gene_summaries","clinvar_20170130","clinvar_20170501","clinvar_20170905","clinvar_20180603","clinvar_20190305","cosmic70","cosmic81","cosmic82","cosmic83_coding","cosmic84_coding","cosmic85_coding","cosmic86_coding","cosmic87_coding","cosmic88_coding","cscd_cancer_circrna","cscd_common_circrna","cscd_normal_circrna","darned","dbnsfp30a","dbnsfp31a_interpro","dbnsfp33a","dbscsnv11","dhs_gene_connectivity_2012","disgenet_gene_disease","disgenet_befree_gene_disease","disgenet_curated_gene_disease","disgenet_rs_disease","disgenet_befree_rs_disease","disgenet_curated_rs_disease","disgenet_pubannotator_rs_disease","docm_3.2","eigen","epi_genes_v1","exac03","exac03nontcga","exac03nonpsych","gdi_score_full_10282015","gme","gnomad_exome","gnomad_genome","gnomad211_exome","gnomad211_genome","gtex_adipose_subcutaneous.v7.egenes","gtex_adipose_visceral_omentum.v7.egenes","gtex_adrenal_gland.v7.egenes","gtex_artery_aorta.v7.egenes","gtex_artery_coronary.v7.egenes","gtex_artery_tibial.v7.egenes","gtex_brain_amygdala.v7.egenes","gtex_brain_anterior_cingulate_cortex_ba24.v7.egenes","gtex_brain_caudate_basal_ganglia.v7.egenes","gtex_brain_cerebellar_hemisphere.v7.egenes","gtex_brain_cerebellum.v7.egenes","gtex_brain_cortex.v7.egenes","gtex_brain_frontal_cortex_ba9.v7.egenes","gtex_brain_hippocampus.v7.egenes","gtex_brain_hypothalamus.v7.egenes","gtex_brain_nucleus_accumbens_basal_ganglia.v7.egenes","gtex_brain_putamen_basal_ganglia.v7.egenes","gtex_brain_spinal_cord_cervical_c_1.v7.egenes","gtex_brain_substantia_nigra.v7.egenes","gtex_breast_mammary_tissue.v7.egenes","gtex_cells_ebv_transformed_lymphocytes.v7.egenes","gtex_cells_transformed_fibroblasts.v7.egenes","gtex_colon_sigmoid.v7.egenes"],["gtex_colon_transverse.v7.egenes","gtex_esophagus_gastroesophageal_junction.v7.egenes","gtex_esophagus_mucosa.v7.egenes","gtex_esophagus_muscularis.v7.egenes","gtex_heart_atrial_appendage.v7.egenes","gtex_heart_left_ventricle.v7.egenes","gtex_liver.v7.egenes","gtex_lung.v7.egenes","gtex_minor_salivary_gland.v7.egenes","gtex_muscle_skeletal.v7.egenes","gtex_nerve_tibial.v7.egenes","gtex_ovary.v7.egenes","gtex_pancreas.v7.egenes","gtex_pituitary.v7.egenes","gtex_prostate.v7.egenes","gtex_skin_not_sun_exposed_suprapubic.v7.egenes","gtex_skin_sun_exposed_lower_leg.v7.egenes","gtex_small_intestine_terminal_ileum.v7.egenes","gtex_spleen.v7.egenes","gtex_stomach.v7.egenes","gtex_testis.v7.egenes","gtex_thyroid.v7.egenes","gtex_uterus.v7.egenes","gtex_vagina.v7.egenes","gtex_whole_blood.v7.egenes","gwava","hgnc_coding_gene2alias","hgnc_coding_pre2gene","hrcr1","icgc21","intervar_20170202","intervar_20180118","intogen_20180119","kaviar_20150923","lncediting","loftool_scores","mcap","nci60","2016sih_wes_ball","2016sih_wes_tall","2016sih_wes_nkt","2016sih_wgs_nkt","2016sih_wgs_dlbcl","2018sih_wgs_nkt_full","2018sih_wgs_nkt_full_sqlite","omim_genemap2_ensembl_gene_id2phenotype","omim_genemap2_entrez_gene_id2phenotype","omim_genemap2_symbol2phenotype","popfreq_max_20150413","popfreq_all_20150413","radar2","rddpred_mes","REDIportal","ucsc_refgene","ens_refgene","revel","rs2pos138","rs2pos142","rs2pos144","rs2pos147","rs2pos150","rvis_exac_4kw","seeqtl_qvalue_hapmap3_cis","seeqtl_qvalue_hapmap3_trans","seeqtl_qvalue_myers_cis","seeqtl_qvalue_myers_trans","tall_somatic_genes_20171206","tmcsnpdb","varcards","vcfanno_demo","vep_all","1000g2015aug_all"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>V1<\/th>\n      <th>V2<\/th>\n      <th>V3<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><html><body><pre class="r"><span class="co"># Get annotation name needed download.name and </span>
<span class="co"># you can use download.database to download database using the download.name.</span>
<span class="no">download.name</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/get.download.name.html">get.download.name</a></span>(<span class="st">'avsnp147'</span>)</pre></body></html>
<html><body><pre class="r"><span class="co"># Database configuration file</span>
<span class="no">database.cfg</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">'extdata'</span>, <span class="st">'config/databases.toml'</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"anor"</span>)

<span class="co"># Get anno.name needed input cols</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/get.annotation.needcols.html">get.annotation.needcols</a></span>(<span class="st">'avsnp147'</span>)
<span class="co">#&gt; [1] "chr"   "start" "end"   "ref"   "alt"</span></pre></body></html>
<p>To facilitate the SQLite format database annotation. anor also provides a simple function <code>sqlite.build</code> to build your sqlite format database: input a text file and output a sqlite format database.</p>
<p>It is helpful for the annotation database with very large size file. Because the SQLite or other SQL format database with the indexes can singlificantly reduce the search and analysis time without any other bio-annotation tools, and just need the powerful SQL database client.</p>
<html><body><pre class="r"><span class="co"># build sqlite database</span>
<span class="kw">for</span>(<span class="no">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"hg19_ALL.sites.2015_08"</span>, <span class="st">"hg19_avsnp147"</span>)) {
  <span class="no">database</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span>(<span class="st">"demo/%s.txt"</span>, <span class="no">i</span>), <span class="kw">package</span> <span class="kw">=</span> <span class="st">"anor"</span>)
  <span class="no">sqlite.db</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span>(<span class="st">"%s/%s.sqlite"</span>, <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="no">i</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/files.html">file.copy</a></span>(<span class="no">database</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span>(<span class="st">"%s/%s.txt"</span>, <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="no">i</span>))
  <span class="fu"><a href="https://rdrr.io/pkg/anor/man/sqlite.build.html">sqlite.build</a></span>(<span class="no">database</span>, <span class="kw">sqlite.connect.params</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">dbname</span> <span class="kw">=</span> <span class="no">sqlite.db</span>, <span class="kw">table.name</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span>(<span class="st">"%s"</span>,
    <span class="no">i</span>)))
}</pre></body></html>
<p>Function <code>annotation</code> is the main interface to access various annotation system including <a href="http://annovar.openbioinformatics.org/en/latest/">ANNOVAR</a>, <a href="https://github.com/brentp/vcfanno">vcfanno</a>, <a href="http://asia.ensembl.org/info/docs/tools/vep/index.html">vep</a>, and R annotation system, such as <a href="http://www.bioconductor.org/packages/devel/bioc/html/AnnotationDbi.html">AnnotationDbi</a> and other R format anntotation script.</p>
<p>For example, if the <code>anno.name</code> contains <code>perl_annovar</code>, it will use the external <code>ANNOVAR</code> to finish the annotation step and read the output file (VCF by <a href="https://CRAN.R-project.org/package=vcfR">vcfR</a>) as the <code>data.table</code> object.</p>
<html><body><pre class="r"><span class="co"># use the defined rule to annotate 1000 Genome Project frequency</span>
<span class="no">database.dir</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>()
<span class="no">chr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr1"</span>)
<span class="no">start</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"10177"</span>, <span class="st">"10177"</span>, <span class="st">"10020"</span>)
<span class="no">end</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"10177"</span>, <span class="st">"10177"</span>, <span class="st">"10020"</span>)
<span class="no">ref</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"-"</span>, <span class="st">"A"</span>, <span class="st">"A"</span>)
<span class="no">alt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"C"</span>, <span class="st">"AC"</span>, <span class="st">"-"</span>)
<span class="no">dat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="kw">chr</span> <span class="kw">=</span> <span class="no">chr</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="no">start</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="no">end</span>, <span class="kw">ref</span> <span class="kw">=</span> <span class="no">ref</span>, <span class="kw">alt</span> <span class="kw">=</span> <span class="no">alt</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">dat</span>, <span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"1000g2015aug_all"</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>, <span class="kw">db.type</span> <span class="kw">=</span> <span class="st">"txt"</span>)
<span class="no">x</span>
<span class="co">#&gt;    1000g2015aug_all</span>
<span class="co">#&gt; 1:         0.425319</span>
<span class="co">#&gt; 2:             &lt;NA&gt;</span>
<span class="co">#&gt; 3:             &lt;NA&gt;</span>
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">dat</span>, <span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"1000g2015aug_all"</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>, <span class="kw">db.type</span> <span class="kw">=</span> <span class="st">"sqlite"</span>)
<span class="no">x</span>
<span class="co">#&gt;    1000g2015aug_all</span>
<span class="co">#&gt; 1:         0.425319</span>
<span class="co">#&gt; 2:             &lt;NA&gt;</span>
<span class="co">#&gt; 3:             &lt;NA&gt;</span></pre></body></html>
</div>
<div id="advanced-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#advanced-annotation" class="anchor"></a>Advanced annotation</h2>
<p>Excepting the predetermined shortcut <code>anno.name</code>, anor also provides the austomizable functions <code>annotation.cols.match</code> and <code>annotation.region.match</code> for the full match and region match respectively.</p>
<html><body><pre class="r"><span class="co"># Do annotation using full match function (default to use chr, start to select data </span>
<span class="co"># and use chr, start, end, ref, and alt to match data)</span>
<span class="co"># Use `?annotation.cols.match` to see more detail about `annotation.cols.match`</span>
<span class="no">chr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr1"</span>)
<span class="no">start</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"10020"</span>, <span class="st">"10020"</span>, <span class="st">"10020"</span>)
<span class="no">end</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"10020"</span>, <span class="st">"10020"</span>, <span class="st">"10020"</span>)
<span class="no">ref</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"A"</span>)
<span class="no">alt</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"-"</span>, <span class="st">"-"</span>, <span class="st">"-"</span>)
<span class="no">dat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="kw">chr</span> <span class="kw">=</span> <span class="no">chr</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="no">start</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="no">end</span>, <span class="kw">ref</span> <span class="kw">=</span> <span class="no">ref</span>, <span class="kw">alt</span> <span class="kw">=</span> <span class="no">alt</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.cols.match.html">annotation.cols.match</a></span>(<span class="no">dat</span>, <span class="st">"avsnp147"</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>,
  <span class="kw">return_col_names</span> <span class="kw">=</span> <span class="st">"avSNP147"</span>, <span class="kw">db.type</span> <span class="kw">=</span> <span class="st">"sqlite"</span>)
<span class="no">x</span>
<span class="co">#&gt;       avSNP147</span>
<span class="co">#&gt; 1: rs775809821</span>
<span class="co">#&gt; 2:        &lt;NA&gt;</span>
<span class="co">#&gt; 3: rs775809821</span>

<span class="co"># Region match mode</span>
<span class="no">bed.file</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"demo/example.bed"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"anor"</span>)
<span class="no">chr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"chr10"</span>, <span class="st">"chr1"</span>)
<span class="no">start</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"100188904"</span>, <span class="st">"100185955"</span>)
<span class="no">end</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"100188904"</span>, <span class="st">"100185955"</span>)
<span class="no">dat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="kw">chr</span> <span class="kw">=</span> <span class="no">chr</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="no">start</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="no">end</span>)

<span class="co"># format.cols.plus.chr will add "chr" in chr colum </span>
<span class="co"># if your input chr colum not contain string 'chr'</span>
<span class="co"># format.db.region.tb will process the region matched data</span>
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.region.match.html">annotation.region.match</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">dat</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="kw">dbname_fixed</span> <span class="kw">=</span> <span class="no">bed.file</span>,
  <span class="kw">table_name_fixed</span> <span class="kw">=</span> <span class="st">"bed"</span>, <span class="kw">db.type</span> <span class="kw">=</span> <span class="st">"txt"</span>, <span class="kw">format_dat_fun</span> <span class="kw">=</span> <span class="st">"format.cols.plus.chr"</span>,
  <span class="kw">format_db_tb_fun</span> <span class="kw">=</span> <span class="st">"format.db.region.tb"</span>)
<span class="no">x</span>
<span class="co">#&gt;      V4</span>
<span class="co">#&gt; 1: HPS1</span>
<span class="co">#&gt; 2: &lt;NA&gt;</span></pre></body></html>
<p>Other various small annotation functions/script also be provided, and the number of the annotation function will continue to increase.</p>
<html><body><pre class="r"><span class="co"># Demo of `convert dbsnp rs number to genomic location` and `alias of human gene symbol`.</span>
<span class="co"># rs2pos: convert snp rs number to genomic location</span>
<span class="no">snp.id</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"rs775809821"</span>, <span class="st">"rs768019142"</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="kw">rs</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="no">snp.id</span>, <span class="fl">3</span>)), <span class="kw">database.dir</span> <span class="kw">=</span> <span class="no">database.dir</span>, <span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"rs2pos147"</span>,
    <span class="kw">buildver</span> <span class="kw">=</span> <span class="st">"hg19"</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">db.type</span> <span class="kw">=</span> <span class="st">"txt"</span>)

<span class="co"># Annotate gene from BioConductor org.hs.eg.db</span>
<span class="no">gene</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"TP53"</span>, <span class="st">"NSD2"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">dat</span> <span class="kw">=</span> <span class="no">gene</span>, <span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"bioc_gene2alias"</span>)
<span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span>
<span class="co">#&gt;     SYMBOL  ALIAS</span>
<span class="co">#&gt;  1:   TP53   BCC7</span>
<span class="co">#&gt;  2:   TP53  BMFS5</span>
<span class="co">#&gt;  3:   TP53   LFS1</span>
<span class="co">#&gt;  4:   TP53    P53</span>
<span class="co">#&gt;  5:   TP53  TRP53</span>
<span class="co">#&gt;  6:   TP53   TP53</span>
<span class="co">#&gt;  7:   NSD2  KMT3F</span>
<span class="co">#&gt;  8:   NSD2  KMT3G</span>
<span class="co">#&gt;  9:   NSD2  MMSET</span>
<span class="co">#&gt; 10:   NSD2 REIIBP</span>
<span class="co">#&gt; 11:   NSD2   TRX5</span>
<span class="co">#&gt; 12:   NSD2    WHS</span>
<span class="co">#&gt; 13:   NSD2  WHSC1</span>
<span class="co">#&gt; 14:   NSD2   NSD2</span></pre></body></html>
</div>
<div id="external-annotation-system" class="section level2">
<h2 class="hasAnchor">
<a href="#external-annotation-system" class="anchor"></a>External annotation system</h2>
<p>If you want to use the ANNOVAR of perl version, you need to download the ANNOVAR source code using R package <a href="https://life2cloud.com/tools/bioinstaller/">BioInstaller</a>, and also need to prepare the <code>avinput</code> format refer the <a href="http://annovar.openbioinformatics.org/en/latest/user-guide/input/#annovar-input-file">tutorial</a>:</p>
<table class="table">
<thead><tr class="header">
<th align="left">chr</th>
<th align="left">start</th>
<th align="left">end</th>
<th align="left">ref</th>
<th align="left">alt</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">chr1</td>
<td align="left">100000</td>
<td align="left">100000</td>
<td align="left">A</td>
<td align="left">T</td>
</tr>
<tr class="even">
<td align="left">chr1</td>
<td align="left">100000</td>
<td align="left">100001</td>
<td align="left">AA</td>
<td align="left">-</td>
</tr>
<tr class="odd">
<td align="left">chr1</td>
<td align="left">100000</td>
<td align="left">100000</td>
<td align="left">-</td>
<td align="left">T</td>
</tr>
<tr class="even">
<td align="left">chr1</td>
<td align="left">100000</td>
<td align="left">100000</td>
<td align="left">-</td>
<td align="left">CAC</td>
</tr>
</tbody>
</table>
<html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">BioInstaller</span>)
<span class="fu"><a href="https://rdrr.io/pkg/BioInstaller/man/install.bioinfo.html">install.bioinfo</a></span>(<span class="st">'annovar'</span>, <span class="no">annovar.dir</span>)</pre></body></html>
<p>To reduce the test time, we set <code>debug</code> to <code>TRUE</code>, and it will returend the command to run ANNOVAR.</p>
<html><body><pre class="r"><span class="co"># Annotate avinput format R data using ANNOVAR</span>
<span class="co"># set debug to TRUE will not to run command</span>
<span class="no">chr</span> <span class="kw">=</span> <span class="st">"chr1"</span>
<span class="no">start</span> <span class="kw">=</span> <span class="st">"123"</span>
<span class="no">end</span> <span class="kw">=</span> <span class="st">"123"</span>
<span class="no">ref</span> <span class="kw">=</span> <span class="st">"A"</span>
<span class="no">alt</span> <span class="kw">=</span> <span class="st">"C"</span>
<span class="no">dat</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html">data.table</a></span>(<span class="no">chr</span>, <span class="no">start</span>, <span class="no">end</span>, <span class="no">ref</span>, <span class="no">alt</span>)
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="no">dat</span>, <span class="st">"perl_annovar_refGene"</span>, <span class="kw">annovar.dir</span> <span class="kw">=</span> <span class="st">"/opt/bin/annovar"</span>,
             <span class="kw">database.dir</span> <span class="kw">=</span> <span class="st">"{{annovar.dir}}/humandb"</span>, <span class="kw">debug</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="co">#&gt; /cluster/home/public/opt/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.3.0/perl-5.30.0-v4c3xbcon5cuhnlq7cq2isdr4nmihvwg/bin/perl /opt/bin/annovar/table_annovar.pl /tmp/RtmpS2fNMS/file5ac727af7b9d0 {annovar.dir}/humandb -buildver hg19  -remove -protocol refGene -operation g -nastring NA</span></pre></body></html>
<p>VCF format files can be processed by <a href="http://annovar.openbioinformatics.org/en/latest/">ANNOVAR</a>, <a href="http://asia.ensembl.org/info/docs/tools/vep/index.html">VEP</a> and <a href="https://github.com/brentp/vcfanno">vcfanno</a>. If you don’t want to read the output files in R, you can set the parameter <code>debug</code> to <code>TRUE</code>, and paste/run the returend command to shell client.</p>
<html><body><pre class="r"><span class="co"># Annotate VCF file using ANNOVAR</span>
<span class="co"># set debug to TRUE will not to run command</span>
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"perl_annovar_ensGene"</span>, <span class="kw">input.file</span> <span class="kw">=</span> <span class="st">"/tmp/test.vcf"</span>,
             <span class="kw">annovar.dir</span> <span class="kw">=</span> <span class="st">"/opt/bin/annovar/"</span>, <span class="kw">database.dir</span> <span class="kw">=</span> <span class="st">"{{annovar.dir}}/humandb"</span>,
             <span class="kw">out</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>(), <span class="kw">vcfinput</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">debug</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="co">#&gt; /cluster/home/public/opt/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.3.0/perl-5.30.0-v4c3xbcon5cuhnlq7cq2isdr4nmihvwg/bin/perl /opt/bin/annovar//table_annovar.pl /tmp/test.vcf {annovar.dir}/humandb -buildver hg19 -out /tmp/RtmpS2fNMS/file5ac7243d22fdc -remove -protocol ensGene -operation g -nastring .  -vcfinput</span>

<span class="co"># Annotation VCF file use VEP</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/vep.html">vep</a></span>(<span class="kw">debug</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="co">#&gt; vep --cache_version 91 --assembly GRCh37 --dir /cluster/home/ljf/.vep --output_file variant_effect_output.txt --cache --offline --everything</span>
<span class="co">#&gt; [1] "vep --cache_version 91 --assembly GRCh37 --dir /cluster/home/ljf/.vep --output_file variant_effect_output.txt --cache --offline --everything "</span>
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"vep_all"</span>, <span class="kw">input.file</span> <span class="kw">=</span> <span class="st">"/tmp/test.vcf"</span>,
             <span class="kw">out</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>(), <span class="kw">debug</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="co">#&gt; vep --cache_version 91 --assembly hg19 --dir /cluster/home/ljf/.vep --output_file /tmp/RtmpS2fNMS/file5ac72542bca1b --input_file /tmp/test.vcf --cache --offline --everything</span>

<span class="co"># Annotation VCF file use vcfanno</span>
<span class="fu"><a href="https://rdrr.io/pkg/anor/man/vcfanno.html">vcfanno</a></span>(<span class="kw">debug</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="co">#&gt; vcfanno_linux64 -p 2 /tmp/Rtmp7z5Wx3/temp_libpath5a6a96c49a08/anor/extdata/demo/vcfanno_demo/conf.toml input.vcf &gt; output.vcf</span>
<span class="co">#&gt; [1] "vcfanno_linux64 -p 2 /tmp/Rtmp7z5Wx3/temp_libpath5a6a96c49a08/anor/extdata/demo/vcfanno_demo/conf.toml input.vcf &gt; output.vcf"</span>
<span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/anor/man/annotation.html">annotation</a></span>(<span class="kw">anno.name</span> <span class="kw">=</span> <span class="st">"vcfanno_demo"</span>, <span class="kw">input.file</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"demo/vcfanno_demo/query.vcf.gz"</span>,
                   <span class="kw">package</span> <span class="kw">=</span> <span class="st">"anor"</span>), <span class="kw">out</span> <span class="kw">=</span> <span class="st">"test.vcf"</span>, <span class="kw">vcfanno</span> <span class="kw">=</span> <span class="st">"/path/vcfanno"</span>, <span class="kw">debug</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="co">#&gt; vcfanno_linux64 -base-path /tmp/Rtmp7z5Wx3/temp_libpath5a6a96c49a08/anor/extdata/demo/vcfanno_demo/ -lua /tmp/Rtmp7z5Wx3/temp_libpath5a6a96c49a08/anor/extdata/demo/vcfanno_demo/custom.lua -p 2 /tmp/Rtmp7z5Wx3/temp_libpath5a6a96c49a08/anor/extdata/demo/vcfanno_demo/conf.toml /tmp/Rtmp7z5Wx3/temp_libpath5a6a96c49a08/anor/extdata/demo/vcfanno_demo/query.vcf.gz &gt; test.vcf</span></pre></body></html>
</div>
<div id="databases-in-anor" class="section level2">
<h2 class="hasAnchor">
<a href="#databases-in-anor" class="anchor"></a>Databases in anor</h2>
<p>Detail about supported annotation database can be found in the another <a href="https://life2cloud.com/tools/anor/articles/database_in_anor.html">vignette</a>.</p>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> on the system on which this document was compiled:</p>
<pre><code>#&gt; R version 3.6.3 (2020-02-29)
#&gt; Platform: x86_64-pc-linux-gnu (64-bit)
#&gt; Running under: CentOS Linux 7 (Core)
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS:   /cluster/home/public/opt/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.3.0/r-3.6.3-uejrtyj747x2sv6bzifcmngnx5adznzp/rlib/R/lib/libRblas.so
#&gt; LAPACK: /cluster/home/public/opt/spack/opt/spack/linux-centos7-skylake_avx512/gcc-8.3.0/r-3.6.3-uejrtyj747x2sv6bzifcmngnx5adznzp/rlib/R/lib/libRlapack.so
#&gt; 
#&gt; locale:
#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] rvest_0.3.5          xml2_1.3.1           stringr_1.4.0       
#&gt; [4] RCurl_1.98-1.2       data.table_1.12.8    anor_1.3            
#&gt; [7] BioInstaller_0.4.0-1
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] nlme_3.1-144         bitops_1.0-6         fs_1.4.1            
#&gt;  [4] usethis_1.6.0        devtools_2.3.0       bit64_0.9-7         
#&gt;  [7] httr_1.4.1           rprojroot_1.3-2      tools_3.6.3         
#&gt; [10] backports_1.1.6      DT_0.13              R6_2.4.1            
#&gt; [13] vegan_2.5-6          DBI_1.1.0            BiocGenerics_0.32.0 
#&gt; [16] mgcv_1.8-31          permute_0.9-5        withr_2.1.2         
#&gt; [19] prettyunits_1.1.1    processx_3.4.2       RMySQL_0.10.20      
#&gt; [22] bit_1.1-15.2         compiler_3.6.3       git2r_0.26.1        
#&gt; [25] cli_2.0.2            Biobase_2.46.0       formatR_1.7         
#&gt; [28] desc_1.2.0           callr_3.4.3          ngstk_0.2.3         
#&gt; [31] rappdirs_0.3.1       pkgdown_1.5.1        digest_0.6.25       
#&gt; [34] rmarkdown_2.1        R.utils_2.9.2        ini_0.3.1           
#&gt; [37] pkgconfig_2.0.3      htmltools_0.4.0      sessioninfo_1.1.1   
#&gt; [40] htmlwidgets_1.5.1    rlang_0.4.5          RSQLite_2.2.0       
#&gt; [43] prettydoc_0.3.1      configr_0.3.4        jsonlite_1.6.1      
#&gt; [46] crosstalk_1.1.0.1    R.oo_1.23.0          magrittr_1.5        
#&gt; [49] Matrix_1.2-18        futile.logger_1.4.3  Rcpp_1.0.4.6        
#&gt; [52] S4Vectors_0.24.3     fansi_0.4.1          ape_5.3             
#&gt; [55] R.methodsS3_1.8.0    stringi_1.4.6        yaml_2.2.1          
#&gt; [58] MASS_7.3-51.5        org.Hs.eg.db_3.10.0  pkgbuild_1.0.6      
#&gt; [61] pinfsc50_1.1.0       grid_3.6.3           blob_1.2.1          
#&gt; [64] parallel_3.6.3       listenv_0.8.0        crayon_1.3.4        
#&gt; [67] RcppTOML_0.1.6       lattice_0.20-38      splines_3.6.3       
#&gt; [70] knitr_1.28           ps_1.3.2             optparse_1.6.4      
#&gt; [73] liteq_1.1.0          codetools_0.2-16     stats4_3.6.3        
#&gt; [76] vcfR_1.10.0          pkgload_1.0.2        futile.options_1.0.1
#&gt; [79] glue_1.4.0           evaluate_0.14        lambda.r_1.2.4      
#&gt; [82] remotes_2.1.1        vctrs_0.2.4          testthat_2.3.2      
#&gt; [85] getopt_1.20.3        future_1.16.0        assertthat_0.2.1    
#&gt; [88] xfun_0.13            viridisLite_0.3.0    AnnotationDbi_1.48.0
#&gt; [91] memoise_1.1.0        IRanges_2.20.2       cluster_2.1.0       
#&gt; [94] globals_0.12.5       ellipsis_0.3.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="life">
  <p>anor was developed by <strong><a href="http://life2cloud.com">Jianfeng</a></strong>. </p>
</div>

<div class="author">
  <p>Developed by Jianfeng Li.</p>
  <p>Site built by <a href="http://pkgdown.r-lib.org">pkgdown</a>.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
